web/src/main/java/com/google/accompanist/web/WebView.kt

```diff
@@ -204,6 +204,12 @@ open class AccompanistWebViewClient : WebViewClient() {
         view: WebView?,
         request: WebResourceRequest?
     ): Boolean {
+        // If the url hasn't changed, this is probably an internal event like
+        // a javascript reload. We should let it happen.
+        if (view?.url == request?.url.toString()) {
```

Comment:
If I'm understanding it correctly, this solution is to allow internal `WebView` reload requests to pass through without overriding loading.

I'm wondering if there's an alternative solution using some sort of unique ID to compare against instead of `url != view.url` to call `view.loadUrl`.

Then, instead of returning `false` here, we'd update the unique ID representing the particular request which would trigger a reload.

That might keep the number of codepaths down, and be closer to a more single-source-of-truth?
Not sure if that would change how we would want to do normal reloading.

Reply:
Not that I could find, you get barely any information about the request. I couldn't even tell it was from a js event in this callback.

Reply:
I was thinking about something like

```kotlin
    request?.let {
        state.content = state.content.withUrl(it.url.toString())
        state.requestId = UUID.randomUUID()
    }
```

And then that `requestId` could act like a `key()` to trigger a reload.